magento/ece-tools
# MCLOUD-5991: Restore additional data in env.php if this file was removed

### Description
If env.php does not exist and DB is full of data, ece-tools runs setup:upgrade, but the deployment is still failed. 


- check existence of crypt key. If this data does not exist (whether in env.php or in .magento.env.yaml) - throw exception that there is not enough data for setup:upgrade or DB is not empty for setup:install and provide appropriate steps to resolve this. Otherwise, continue to run Magento upgrade process as it exists now.
- if env.php did not exist or had only one cache type in it - restore all default types, otherwise keep them as they are (we cannot restore cache type data by default as customer could remove some types for purpose)
- add warning that env.php did not exist at the beginning of deployment.

### Fixed Issues (if relevant)

1. https://jira.corp.magento.com/browse/MCLOUD-5991


### Manual testing scenarios

1. on installed Magento remove env.php and run redeployment 
**Expected result:** deployment is failed with message that `crypt key` does not exist
2. remove env.php again (as it was partially restored form prev. deployment) add CRYPT_KEY variable and run redeployment
**Expected result:**
- deployment finished without errors
- env.php has whole list of `cache_types`
- cloud.log contain message that cache_types was restored and that env.php did not exist at the beginning of the deployment
3. in env.php remove a few types in `cache_type` option and set `0` to other cache_types and left few with value `1`. Run redeployment.
**Expected result:** deployment finished. env.php has the same state of `cache_types` option (few removed types, and types with values as 0 and 1)

### Contribution checklist
